_resources/port1.0/group/python-1.0.tcl

```diff
@@ -447,3 +488,33 @@ pre-livecheck {
         livecheck.type  none
     }
 }
+
+# set-up a test phase
+pre-test {
+    if {[option python.test]} {
+        # return for py-<portname> stub ports, which should not have a test-phase enabled
+        if {[string match py-* [option name]] && ${name} eq ${subport}} {
+            return
+        }
+
+        test.run            yes
+        default test.target {}
```

Comment:
All of these defaults need to be set outside of a phase. Within a phase (like a `pre-test` block) is too late for a port to be able to override them.

Reply:
It would also work to set defaults in an option_proc, in case they need to change based on `python.test_framework`.

Reply:
Yeah, thanks for pointing out these issues. I've added `test.args` to the `python_get_defaults` proc and set the default for the `test.target` outside of a phase (I haven't come across a situation where the `test.target` depends on `python.test_framework`.